modules/oe_corporate_countries_address/tests/src/Kernel/AvailableCountriesSubscriberTest.php

```diff
@@ -0,0 +1,310 @@
+
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'user',
+    'system',
+    'field',
+    'text',
+    'filter',
+    'entity_test',
+    'language',
+    'address',
+    'oe_corporate_countries',
+    'oe_corporate_countries_address',
+  ];
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The state service.
+   *
+   * @var \Drupal\Core\State\StateInterface
+   */
+  protected $state;
+
+  /**
+   * Name of the field.
+   *
+   * @var string
+   */
+  protected $fieldName = 'field_country';
+
+  /**
+   * Name of the field type.
+   *
+   * @var string
+   */
+  protected $fieldTypeName = 'address_country';
+
+  /**
+   * Name of the entity type.
+   *
+   * @var string
+   */
+  protected $entityTypeName = 'entity_test';
+
+  /**
+   * Name of the bundle to be tested.
+   *
+   * @var string
+   */
+  protected $bundleName = 'oe_countries';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installEntitySchema('configurable_language');
+    $this->installConfig(['language']);
+    // At least one extra language needs to be enabled for translations to work.
+    ConfigurableLanguage::createFromLangcode('it')->save();
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+    $this->state = $this->container->get('state');
+
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('entity_test');
+
+    $entity = EntityTest::create();
+
+    FieldStorageConfig::create([
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'type' => $this->fieldTypeName,
+    ])->save();
+
+    FieldConfig::create([
+      'label' => 'My field',
+      'field_name' => $this->fieldName,
+      'entity_type' => $this->entityTypeName,
+      'bundle' => $this->entityTypeName,
+    ])->save();
+
+    $entity_form_display = EntityFormDisplay::collectRenderDisplay($entity, 'default');
+
+    $entity_form_display->setComponent($this->fieldName, [
+      'region' => 'content',
+      'type' => 'address_country_default',
+    ]);
+    $entity_form_display->save();
+
+  }
+
+  /**
+   * Tests the deprecated countries are removed from the countries array.
+   * */
+  public function testGetAvailableCountriesArray(): void {
+
+    $values = [
+      $this->fieldName => [
+        'value' => 'FR',
+      ],
+    ];
+
+    $entity = EntityTest::create($values);
+    $entity->save();
+
+    $entity = $this->reloadEntity($entity);
+
+    /** @var \Drupal\field\Entity\FieldStorageConfig $field_storage */
+    $field_storage = $entity->getFieldDefinitions()[$this->fieldName]
+      ->getFieldStorageDefinition();
+
+    $field_storage->getPropertyDefinitions();
+
+    // Get available_countries from field.
+    // This doesn't have the deprecated countries.
+    $form = $this->buildEntityTestForm();
```

Comment:
We don't really need to build a whole entity form to test the event. Doing this does the same job:
$entity = EntityTest::create();
$available_countries = $entity->get('field_country')->appendItem()->getAvailableCountries();